### DISCUSSION TITLE
urban speed limit in UK

### DISCUSSION THREAD

**Discussion by karussell** [westnordost/osm-legal-default-speeds#15] [Q&A]
In UK there is no separate rule for "urban" instead [there is](https://wiki.openstreetmap.org/wiki/Default_speed_limits) a "restricted road" which is `lit=yes or maxspeed:type=GB:nsl_restricted`. The problem is that the latter tag is used only 17k times world wide and lit=yes has not a good coverage too:

![image](https://github.com/westnordost/osm-legal-default-speeds/assets/129644/f1c443b3-a324-426f-8df1-db754ebb1621)

I know it is not in the legal wording but would it make sense to broaden the scope of this rule to e.g. `lit=yes or maxspeed:type=GB:nsl_restricted or urban` in the wiki or would you recommend to better retrieve the urban maxspeed via some [additional client side code](https://github.com/graphhopper/graphhopper/commit/cea958ffdaa05e3f0435bc6718f024d8547ba5b1#diff-79302b5356e2f2d2a187f5c82f00d3115dbe692a9e71dd7f03e1b68a77e2e53eR48)?

btw: is the rule `50 mph (trailer)` in the wiki correct for "restricted road"? It looks wrong as it is higher than without a trailer.

**Comment by westnordost** [comment]
But you won't find `source:maxspeed=GB:urban` in the wild. So, what good would that do?

> btw: is the rule 50 mph (trailer) in the wiki correct for "restricted road"? It looks wrong as it is higher than without a trailer.

Dunno, you will have to check the source linked.

**Reply by karussell** [reply]
> So, what good would that do?

Not sure. Probably I'm asking the wrong question :)

I need a way to query the urban and rural maxspeed. The reason is that at this point of the data import we have all tags available but don't know yet if the OSM way is rural or not. If we would do it later when we know if the way is rural or not then we would not have all tags available.

**Comment by arrival-spring** [comment]
> I know it is not in the legal wording but would it make sense to broaden the scope of this rule to e.g. lit=yes or maxspeed:type=GB:nsl_restricted or urban

I think it would make sense to add {urban} to fuzzy matching, as that would then allow a fuzzy matching of urban roads (living street, residential, sidewalk). There are many residential roads with no additional tags, and which are in reality restricted roads, but would not be recognised as such at the moment.

> > btw: is the rule 50 mph (trailer) in the wiki correct for "restricted road"? It looks wrong as it is higher than without a trailer.
> 
> Dunno, you will have to check the source linked.

I think it is correct, because if you removed it you would also have to remove `20 mph (2 trailers)`, but it seems that both provisions apply to all "other roads" (not motorway or dual carriageway), and then there is the additional restriction of 30 mph for restricted roads.

It doesn't matter so much though because [Always the lowest speed is the applicable one](https://wiki.openstreetmap.org/wiki/Default_speed_limits#Speed_limits_by_vehicle_type)

**Reply by westnordost** [reply]
Ah, now I understand.

Why only for fuzzy though, isn't "restricted road" == "urban"? In which case **isn't** it the same? I am asking because the (fuzzy) filter should contain this too. If `{urban}` just matched "restricted road" (fuzzily or not), then cases where it isn't the same would still be matched. If anything, the (fuzzy) filter would need to be

`{urban} and <is not explicitly NOT "restricted">`

**Reply by arrival-spring** [reply]
> isn't "restricted road" == "urban"? In which case **isn't** it the same?

[Restricted road](https://www.legislation.gov.uk/ukpga/1984/27/part/VI?view=plain) means it has street lamps not more than 200 yards apart (there is no legal definition of 'urban' that I am aware of). Hmm, or a road that the traffic authority decides is a restricted road. You never see signs to the effect though, signs are always explicit (30 mph) for such places, and on restricted roads, signs are in effect until you see another one that changes things.

I do know of some villages where there are no street lights (a dark sky area or something) and so the roads there would not be restricted roads. However they _likely_ have an explicit speed limit, probably 30 mph, but could also be a 20 mph zone.

So fuzzy filter should be something like `{urban} and (!lit or lit!=no)`, or maybe just `{urban} and !lit` since if `lit=yes` then it hits the exact match.

**Reply by westnordost** [reply]
My point is

```
sidewalk=both
maxspeed:type=GB:nsl_single
```
(`sidewalk=both` is a fuzzy match for "urban").

This would then match to "restricted road".

**Reply by arrival-spring** [reply]
Is an exact match not preferred to a fuzzy match though? That would exact match "rural" and only fuzzy match "restricted road".

**Reply by westnordost** [reply]
That's details of that specific example and the problem would only not trigger due to the intricacies of the match logic. in that case.

Better if the filter is written in a way that `maxspeed:type=GB:nsl_single` will **never** match to "restricted road", fuzzy or not, because that tag explicitly states that it is not a restricted road. 

So, anyway, for `{urban} and <is not explicitly NOT "restricted">`, what would be the proper filter for the `<....>`?

**Reply by arrival-spring** [reply]
I think `(!lit or lit!=no) and !maxspeed:type` would do it
